assets/js/modules/idea-hub/datastore/idea-state.js

```diff
@@ -280,8 +247,69 @@ const baseActions = {
 			type: REMOVE_ACTIVITIES,
 		};
 	},
+
+	/**
+	 * Moves an idea from the list of newIdeas state variable to savedIdeas state variable.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} name Idea name.
+	 */
+	*moveIdeaFromNewIdeasToSavedIdeas( name ) {
+		const idea = yield findIdeaByName( name, 'newIdeas' );
+		yield removeIdeaFromList( idea, 'newIdeas' );
+		yield addIdeaToList( idea, 'savedIdeas' );
+	},
+
+	/**
+	 * Moves an idea from the list of savedIdeas state variable to newIdeas state variable.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} name Idea name.
+	 */
+	*moveIdeaFromSavedIdeasToNewIdeas( name ) {
+		const idea = yield findIdeaByName( name, 'savedIdeas' );
+		yield removeIdeaFromList( idea, 'savedIdeas' );
+		yield addIdeaToList( idea, 'newIdeas' );
+	},
+
+	/**
+	 * Removes an idea from the list of newIdeas state variable.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} name Idea name.
+	 */
+	*removeIdeaFromNewIdeas( name ) {
+		const idea = yield findIdeaByName( name, 'newIdeas' );
+		yield removeIdeaFromList( idea, 'newIdeas' );
+	},
 };
 
+// Utility action for selecting `getIdeaByName`.
+// In newer versions of WP data, this is doable using
+// the `select` data control.
+function* findIdeaByName( name, list ) {
+	const { select } = yield Data.commonActions.getRegistry();
+
+	return select( MODULES_IDEA_HUB ).getIdeaByName( name, list );
+}
+
+function addIdeaToList( idea, list ) {
+	return {
+		payload: { idea, list },
+		type: ADD_IDEA_TO_LIST,
+	};
+}
+
+function removeIdeaFromList( idea, list ) {
+	return {
+		payload: { idea, list },
+		type: REMOVE_IDEA_FROM_LIST,
+	};
+}
+
```

Comment:
These functions should be actions. Please, move them to the corresponding baseActions objects.

Reply:
These are low-level utility actions that we don't particularly want/need to expose externally. This way we can still use them for our needs here without making them into fully bound action creators.

Reply:
> we don't particularly want/need to expose externally

@aaemnnosttv why? We have similar utility actions in other modules defined as regular actions. I think there is nothing wrong in keeping these utility functions as actual actions.

Reply:
They're not any less actions as `baseActions`, they're just not registered as bound action creators (i.e. `dispatch(...)`). We have quite a few of these in the codebase but they're using an inline literal form (search for `yield {`) – these just have a function.

I'm not strongly opposed to adding them to be dispatchable like the others, but it isn't a novel concept/pattern in the datastore. If we were to move them then they should also get tests for each which is something every bound action should get. Right now they're covered implicitly by the higher actions that use them, so it isn't as if they're untested.

Thoughts @felixarntz ?